provisioner/shell/provisioner_acc_test.go

```diff
@@ -1,41 +1,77 @@
 package shell_test
 
 import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/packer/helper/tests/acc"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/hashicorp/packer/packer"
+
 	"github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/packer/command"
 	testshelper "github.com/hashicorp/packer/helper/tests"
 )
 
-func TestBuildShellProvisionerWithBuildVariablesSharing(t *testing.T) {
-	testshelper.AccTestPreValidate(t)
+func TestShellProvisioner(t *testing.T) {
+	p := os.Getenv("ACC_TEST_PROVISIONERS")
+	if p != "all" && !strings.Contains(p, "shell") {
+		t.Skip()
+	}
+	acc.TestProvisionersAgainstBuilders(new(ShellProvisionerAccTest), t)
```

Comment:
Moved the tests to provisioners and now the helper will receive the provisioner data and run builders against it

Reply:
This is nice because we don't need a provisioner list in the generic logic and the helper package doesn't need to know the provisioner

Reply:
I like this a lot.